### PR TITLE
feat: random builder

### LINKED ISSUE #12171: Create Random Number Manager
Create a platform utility that acts as the source of all randomness within the platform. This utility should be used whenever we construct `Random` objects (or their cryptographically secure equivalents). 

The purpose of this utility is to provide a unit test hook where we can provide a seed. This is a prerequisite for deterministic TURTLE testing.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #12171

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SyncManagerTest.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public SyncManagerTestData() {
             this.selfId = addressBook.getNodeId(0);
             final int size = addressBook.getSize();
 
-            connectionGraph = new RandomGraph(size, 40, 0);
+            connectionGraph = new RandomGraph(new Random(), size, 40, 0);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SyncManagerTest.java:76]: Should use the `random` already created above

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java [hunk 2]
```diff
@@ -57,7 +57,7 @@ void testShouldConnectToMe(final int numNodes, final int numNeighbors) throws Ex
         final AddressBook addressBook =
                 new RandomAddressBookGenerator(r).setSize(numNodes).build();
         final NodeId selfId = addressBook.getNodeId(r.nextInt(numNodes));
-        final StaticTopology topology = new StaticTopology(addressBook, selfId, numNeighbors);
+        final StaticTopology topology = new StaticTopology(new Random(), addressBook, selfId, numNeighbors);
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java:60]: use `random` already created

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java [hunk 3]
```diff
@@ -90,7 +90,7 @@ void testShouldConnectTo(final int numNodes, final int numNeighbors) throws Exce
...
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java:93]: use `random` already created

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java [hunk 4]
```diff
@@ -107,7 +107,7 @@ private static void testRandomGraphWithSets(
     @MethodSource({"failing", "topologicalVariations", "fullyConnected"})
     void testRandomGraphs(final int numNodes, final int numNeighbors, final long seed) throws Exception {
         System.out.println("numNodes = " + numNodes + ", numNeighbors = " + numNeighbors + ", seed = " + seed);
-        final RandomGraph randomGraph = new RandomGraph(numNodes, numNeighbors, seed);
+        final RandomGraph randomGraph = new RandomGraph(new Random(), numNodes, numNeighbors, seed);
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java:110]: print random seed?

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java [hunk 5]
```diff
@@ -121,7 +121,7 @@ void testFullyConnectedTopology(final int numNodes, final int numNeighbors, fina
         for (int thisNode = 0; thisNode < numNodes; thisNode++) {
             final NodeId outOfBoundsId = addressBook.getNextNodeId();
             final NodeId thisNodeId = addressBook.getNodeId(thisNode);
-            final NetworkTopology topology = new StaticTopology(addressBook, thisNodeId, numNeighbors);
+            final NetworkTopology topology = new StaticTopology(new Random(), addressBook, thisNodeId, numNeighbors);
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java:124]: print random seed?

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/reconnect/FallenBehindManagerTest.java [hunk 6]
```diff
@@ -38,7 +39,7 @@ class FallenBehindManagerTest {
             new RandomAddressBookGenerator().setSize(numNodes).build();
     final double fallenBehindThreshold = 0.5;
     final NodeId selfId = addressBook.getNodeId(0);
-    final RandomGraph graph = new RandomGraph(numNodes, numNodes + (numNodes % 2), numNodes);
+    final RandomGraph graph = new RandomGraph(new Random(), numNodes, numNodes + (numNodes % 2), numNodes);
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/reconnect/FallenBehindManagerTest.java:42]: print random seed?

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java:60]: done

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java:93]: done

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java:110]: Agreed, that is a better pattern. Change made.

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java:124]: changed

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/reconnect/FallenBehindManagerTest.java (hunk 6 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/reconnect/FallenBehindManagerTest.java:42]: change made

**Review by alittley — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Pull request merged**